packages/ic-asset-certification/src/asset_router.rs

```diff
@@ -117,23 +117,39 @@ struct CertifiedAssetResponse<'a> {
 #[derive(Debug)]
 pub struct AssetRouter<'content> {
     tree: Rc<RefCell<HttpCertificationTree>>,
-    responses: HashMap<String, CertifiedAssetResponse<'content>>,
-    encoded_responses: HashMap<(String, String), CertifiedAssetResponse<'content>>,
-    fallback_responses: HashMap<String, CertifiedAssetResponse<'content>>,
-    encoded_fallback_responses: HashMap<(String, String), CertifiedAssetResponse<'content>>,
+    responses: HashMap<RequestKey, CertifiedAssetResponse<'content>>,
+    fallback_responses: HashMap<RequestKey, CertifiedAssetResponse<'content>>,
+}
+
+/// A key created from request data, to retrieve the corresponding response.
+#[derive(Debug, Eq, Hash, PartialEq)]
+pub struct RequestKey {
+    /// Path of the requested asset.
+    pub path: String,
+    /// The encoding of the assset.
+    pub encoding: Option<String>,
+}
+
+fn request_key(path: &str, encoding: Option<String>) -> RequestKey {
+    RequestKey {
+        path: path.to_string(),
+        encoding,
+    }
 }
 
 const IC_CERTIFICATE_EXPRESSION_HEADER: &str = "IC-CertificateExpression";
 
+fn encoding_str(maybe_encoding: Option<AssetEncoding>) -> Option<String> {
```

Comment:
Could we co-locate this with the other code for `AssetEncoding` in the `asset_config.rs` file?

Reply:
Same here, I'll do this change in the next PR.